src/ZkMinterERC1155V1.sol

```diff
@@ -0,0 +1,186 @@
+  /// interface can be used, but in most cases this will be another `ZKMinter` extension or `ZKCappedMinter`.
+  /// @param _admin The address that will have admin privileges.
+  /// @param _erc1155 The ERC1155 contract to use for minting.
+  /// @param _tokenId The token id for the ERC1155 contract.
+  /// @param _balanceThreshold The balance threshold for the ERC1155 contract.
+  constructor(IMintable _mintable, address _admin, address _erc1155, uint256 _tokenId, uint256 _balanceThreshold) {
+    if (_admin == address(0)) {
+      revert ZkMinterERC1155V1__InvalidZeroAddress();
+    }
+
+    _updateMintable(_mintable);
+    _updateERC1155(_erc1155);
+    _updateTokenId(_tokenId);
+    _updateBalanceThreshold(_balanceThreshold);
+
+    _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+    _grantRole(PAUSER_ROLE, _admin);
+  }
+
+  /*///////////////////////////////////////////////////////////////
+                          External Functions
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Allows a caller to mint tokens to a specified address if they meet the ERC1155 balance threshold.
+  /// @param _to The address to mint the tokens to.
+  /// @param _amount The amount of tokens to mint.
+  /// @dev Caller must hold at least `balanceThreshold` of token `tokenId` in `erc1155`.
+  function mint(address _to, uint256 _amount) external {
+    _revertIfClosed();
+    _requireNotPaused();
+    _revertIfInsufficientBalance();
+
+    if (_to == address(0)) {
+      revert ZkMinterERC1155V1__InvalidZeroAddress();
+    }
+
+    if (_amount == 0) {
+      revert ZkMinterERC1155V1__InvalidAmount();
+    }
+
+    // mint the tokens
+    mintable.mint(_to, _amount);
+
+    emit Minted(msg.sender, _to, _amount);
+  }
+
+  /// @notice Updates the ERC1155 contract for the ZkMinter.
+  /// @param _erc1155 The ERC1155 contract to use for balance checks.
+  function updateERC1155(address _erc1155) external {
+    _checkRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    _updateERC1155(_erc1155);
+  }
+
+  /// @notice Updates the token id for the ERC1155 contract.
+  /// @param _tokenId The new token id to set.
+  function updateTokenId(uint256 _tokenId) external {
+    _checkRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    _updateTokenId(_tokenId);
+  }
+
+  /// @notice Updates the balance threshold for the ERC1155 contract.
+  /// @param _balanceThreshold The new balance threshold to set.
+  /// @dev This is the minimum balance that must be held by the caller to mint.
+  function updateBalanceThreshold(uint256 _balanceThreshold) external {
+    _checkRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    _updateBalanceThreshold(_balanceThreshold);
+  }
+
+  /// @notice Returns whether the given address is eligible to mint.
+  /// @param _caller The address to check eligibility for.
+  /// @return True if the caller has a balance greater than or equal to the balance threshold, false otherwise.
+  function isEligible(address _caller) public view returns (bool) {
```

Comment:
@alexkeating not in spec but figured it might be a nice util function for UI / UX ?